### DISCUSSION TITLE
Feature: create output based on ActivityPub protocol

### DISCUSSION THREAD

**Discussion by tosteiner** [knowledgefutures/pubpub#2339] [Ideas]
This might be a bit of a geeky idea, but I was wondering if PubPub could be configured to output content that would be conformant with ActivityPub ...

With the recent surge in interest of academics to join the Fediverse, particularly Mastodon, caused by what's been happening on the birdsite, this might become more and more relevant as an ecosystem to consider, and since [ActivityPub](https://www.w3.org/TR/activitypub/) is the W3C-approved protocol that lies underneath all of that, it might be be interesting for PubPub to look into making (versions of) Pubs compatible with that?

I noticed there's work being done over at WordPress on plugins that support an integration of ActivityPub, see e.g. https://wordpress.org/plugins/activitypub/ - maybe this can be of use?

EDIT: This might also nicely link up to #1386 , and connects to the amazing research done by Sarven Capadisli ( @csarven ) on the decentralised web, see https://csarven.ca/linked-research-decentralised-web and, most notably, his work on [Dokieli](https://dokie.li/).

**Comment by gabestein** [comment]
As the proprietor of gabestein.com, I'm very into the idea. Putting on my PubPub PM hat, it might not be the highest-priority thing, but certainly something worth looking into and seeing what we can support, especially because it might not be all that difficult.

I would like to know more, for folks who are exploring the Fediverse, what functionality would be useful. Automatically pushing Pubs to Mastodon et. al. and collecting replies from them as discussions seems like the obvious thing. The proprietor of GabeStein.com would certainly like that, because he doesn't like to actively use social media, but wants to be able to participate in discussions around longer-form content. But as I'm not an active user of much social media at all these days, and have never really delved deeply in the Fediverse, it would be useful to better understand what folks might want out of it.

**Reply by gobengo** [reply]
* As a learner, I want to follow pubpubs using my social web client, so I can learn about new issues/articles using a news reader of my choice.
* As a learner, I want to reply to a pubpub article using my social web client, so I can ask the author a question or to point out a typo or other mistake

**Comment by tosteiner** [comment]
Hi Gabe, hi all, 

Gabe thanks so much for your follow up, and apologies for the radio silence - in my view, the beauty of ActivityPub really is that it's app-independent / not tied to a single use case / app, i.e. when I make my WordPress blog content available via ActivityPub, it's not only Mastodon readers who might pick up on it, but users of any application supporting AP. (For more on AP, see [this recent interview with Evan Prodromou]((https://thenewstack.io/the-creator-of-activitypub-on-whats-next-for-the-fediverse/) ), the protocol's creator ... )

Mike Thicke (Humanities Commons) has written a brilliant summary of the trajectory that hCommons is seeking to put in place with their pursuing a "federation of networks" strategy via ActivityPub as the underlying protocol, which I found really inspiring, and there's many overlaps with my own thoughts on this, so I thought I'd post it here for reference :) https://building.hcommons.org/2022/12/02/a-federated-commons/

And just to echo Gabe's prompt, I would similarly be interested in hearing more about other folks' expectations with regards to ActivityPub (if any?) 😃

**Comment by tosteiner** [comment]
Just stumled over this recent piece on The Verge, which might be of interest here:  https://www.theverge.com/2023/4/20/23689570/activitypub-protocol-standard-social-network

**Comment by gobengo** [comment]
Here is a past discussion on the topic https://github.com/pubpub/pubpub/discussions/1249
